storage/backend/ceph/internals.go

```diff
@@ -1,26 +1,32 @@
 package ceph
 
 import (
+	"encoding/json"
 	"fmt"
 	"os/exec"
 	"regexp"
 	"strings"
 
-	"encoding/json"
-
 	"github.com/contiv/volplugin/storage"
 
 	log "github.com/Sirupsen/logrus"
 )
 
 func (c *Driver) mapImage(do storage.DriverOptions) (string, error) {
-	var rbdMaps map[string] struct {
-		Pool string `json:"pool"`
-		Name string `json:"name"`
+	var rbdMaps map[string]struct {
+		Pool   string `json:"pool"`
+		Name   string `json:"name"`
 		Device string `json:"device"`
 	}
 
-	blkdev, err := exec.Command("rbd", "map", do.Volume.Name, "--pool", do.Volume.Params["pool"]).Output()
+	poolName := do.Volume.Params["pool"]
+
+	if err := exec.Command("rbd", "lock", "add", do.Volume.Name, do.Volume.Name, "--pool", poolName).Run(); err != nil {
+		log.Debugf("Could not acquire lock for %q: %v", do.Volume.Name, err)
+		return "", err
+	}
+
```

Comment:
just wondering if we shall defer a unlock operation in case one of the operations later fail?